angularjs-portal-frame/src/main/webapp/main.js

```diff
@@ -0,0 +1,59 @@
+require.config({
+
+  paths: {
+    'my-app': 'my-app.js'
+    /* TODO: Incorporate the <scripts/> that are being globally imported in frame.html

```

Comment:
I'd convert our own code to require modules, and most of the libraries need shim configurations, which just means going through one by one to find what they depend on and what they export.  It's mostly rote, but there might need to be changes in how things initialize, like I had to do here for angular.  I've heard that Google Analytics can break inside the async loading of require.js without some change, but I didn't dig in to see how much effort is needed.